lib/src/js/compiler.dart

```diff
@@ -0,0 +1,104 @@
+import 'dart:js_util';
+
+import 'package:node_interop/js.dart';
+
+import 'compile.dart';
+import 'compile_options.dart';
+import 'reflection.dart';
+import 'utils.dart';
+
+class Compiler {
+  bool _disposed = false;
+
+  void throwIfDisposed() {
+    if (_disposed) {
+      jsThrow(JsError('Compiler has already been disposed.'));
+    }
+  }
+}
+
+class AsyncCompiler extends Compiler {
+  final Set<Promise> _compilations = {};
+
+  /// Adds a compilation to the pending set and removes it when it's done.
+  void _addCompilation(Promise compilation) {
+    _compilations.add(compilation);
+    compilation.then((value) {
+      _compilations.remove(compilation);
+    }, (error) {
+      _compilations.remove(compilation);
+    });
+  }
+}
+
+/// The JS API Compiler class
+///
+/// See https://github.com/sass/sass/spec/tree/main/js-api/compile.d.ts for
```

Comment:
This link isn't working for me- I think this should be https://github.com/sass/sass/blob/main/spec/js-api/compile.d.ts.md. Although I do wonder, since the complexity of the Compiler is much less than `compile`, if it makes sense to inline these descriptions.

Reply:
You're right, the other JS classes don't have links to the spec. Updated in e6ff49b3a05882c8a935466e461e8aa43d1c945c